src/coreclr/interpreter/compiler.cpp

```diff
@@ -807,6 +857,34 @@ void InterpCompiler::EmitCode()
     m_compHnd->setBoundaries(m_methodInfo->ftn, m_ILToNativeMapSize, m_pILToNativeMap);
 }
 
+void InterpCompiler::BuildGCInfo(InterpMethod *pInterpMethod)
+{
+#ifdef FEATURE_INTERPRETER
+    InterpIAllocator* pAllocator = new InterpIAllocator(this);
+    InterpreterGcInfoEncoder* gcInfoEncoder = new InterpreterGcInfoEncoder(m_compHnd, m_methodInfo, pAllocator, Interp_NOMEM);
+    assert(gcInfoEncoder);
+
+    gcInfoEncoder->SetCodeLength(m_methodCodeSize);
+
+    // TODO: Request slot IDs for all our locals before finalizing
+
+    gcInfoEncoder->FinalizeSlotIds();
+
+    // TODO: Use finalized slot IDs to declare live ranges
+
+    gcInfoEncoder->Build();
+
+    // GC Encoder automatically puts the GC info in the right spot using ICorJitInfo::allocGCInfo(size_t)
+    // let's save the values anyway for debugging purposes
+    gcInfoEncoder->Emit();
+
+    // Interpreter-FIXME: Why doesn't the JIT code do this? Is it because the placement new it uses automatically frees them
+    //  at the end of JIT compilation?
```

Comment:
Yes. To be precise, it doesn't use placement new, but a custom operator new defined here:
https://github.com/dotnet/runtime/blob/ff18c306e5878215f9cefdd3c72e91e48826f4f0/src/coreclr/jit/compiler.hpp#L4614
It allocates from an arena allocator and the whole arena is freed at the end of the compilation. In the interpreter compiler, Vlad has introduced the AllocMemPool0 that is supposed to free the memory when the compilation of a method is done. It is implemented as a malloc now, but the plan is to make it proper. So you could use that + placement new.